.github/workflows/ci-release.yml

```diff
@@ -0,0 +1,65 @@
+name: ci-release
+
+on:
+  push:
+    branches:
+      - main
+
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+
+    defaults:
+      run:
+        shell: bash -l {0}
+
+    env:
+      WF_VERSION_FNAME: "version.txt"
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: "calculate release version"
+        run: |
+          git pull
+          if [ -f "${WF_VERSION_FNAME}" ]
+          then
+            WF_VERSION=$(cat ${WF_VERSION_FNAME})
+          else
+            WF_VERSION="$(date +%Y.%m).0"
+            echo "WF_BOOTSTRAP=1" >> ${GITHUB_ENV}
+          fi
+          WF_DATE=$(echo ${WF_VERSION} | cut -d. -f1-2)
+          WF_SERIAL=$(echo ${WF_VERSION} | cut -d. -f3)
+          NOW_DATE=$(date +%Y.%m)
+          NOW_SERIAL=0
+          if [ "${WF_DATE}" == "${NOW_DATE}" ]
+          then
+            NOW_SERIAL=$(expr ${WF_SERIAL} + 1)
```

Comment:
Does this not mean that bootstrap mode would always start on 1?

If `version.txt` doesn't exist, `WF_VERSION` is being set to `$(date +%Y.%m).0`, so `WF_DATE` is effectively `$(date +%Y.%m)` which is the same as `NOW_DATE` so this is statement `WF_DATE == NOW_DATE` will always be true.

I think it would make sense to add these lines into the if block above, as they only apply when there is an existing `version.txt` file, i.e. 
```
if [ -f "${WF_VERSION_FNAME}" ]
then
  WF_VERSION=$(cat ${WF_VERSION_FNAME})
  WF_DATE=$(echo ${WF_VERSION} | cut -d. -f1-2)
  WF_SERIAL=$(echo ${WF_VERSION} | cut -d. -f3)
  NOW_DATE=$(date +%Y.%m)
  if [ "${WF_DATE}" == "${NOW_DATE}" ]
  then
    WF_SERIAL=$(expr ${WF_SERIAL} + 1)
  fi
  WF_VERSION=${NOW_DATE}.${WF_SERIAL}
else
  WF_VERSION="$(date +%Y.%m).0"
  echo "WF_BOOTSTRAP=1" >> ${GITHUB_ENV}
echo "WF_VERSION=${WF_VERSION}" >> ${GITHUB_ENV}
```